int32 a
int32 b
int64 sum_ab
int32 diff_ab
int64 product_ab
int32 quotient_ab
int32 remainder_ab